consensus/polybft/state_sync_manager.go

```diff
@@ -0,0 +1,469 @@
+package polybft
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"path"
+	"sync"
+
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/bitmap"
+	polybftProto "github.com/0xPolygon/polygon-edge/consensus/polybft/proto"
+	bls "github.com/0xPolygon/polygon-edge/consensus/polybft/signer"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/wallet"
+	"github.com/0xPolygon/polygon-edge/tracker"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/umbracle/ethgo"
+	"google.golang.org/protobuf/proto"
+)
+
+// stateSyncConfig holds the configuration data of state sync manager
+type stateSyncConfig struct {
+	stateSenderAddr types.Address
+	jsonrpcAddr     string
+	dataDir         string
+	topic           topic
+	key             *wallet.Key
+}
+
+// StateSyncManager is a struct that manages the workflow of
+// saving and querying state sync events, and creating, and submitting new commitments
+type StateSyncManager struct {
+	// configuration fields
+	logger hclog.Logger
+	state  *State
+
+	config *stateSyncConfig
+
+	// per epoch fields
+	lock         sync.Mutex
```

Comment:
Could this be `sync.RWMutex`? It is there only because of validator set basically if I'm correct?

Reply:
It will be changed in another PR, for multiple commitments?

Reply:
Good for me